snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedTransientSnapshot.java

```diff
@@ -48,28 +49,34 @@
   }
 
   @Override
-  public ActorFuture<Boolean> take(final Predicate<Path> takeSnapshot) {
+  public ActorFuture<Void> take(final Consumer<Path> takeSnapshot) {
     actor.run(() -> takeInternal(takeSnapshot));
     return takenFuture;
   }
 
-  private void takeInternal(final Predicate<Path> takeSnapshot) {
+  private void takeInternal(final Consumer<Path> takeSnapshot) {
     final var snapshotMetrics = snapshotStore.getSnapshotMetrics();
 
     try (final var ignored = snapshotMetrics.startTimer()) {
       try {
-        isValid = takeSnapshot.test(getPath());
-        if (!isValid) {
-          abortInternal();
-        } else if (!directory.toFile().exists() || directory.toFile().listFiles().length == 0) {
+        takeSnapshot.accept(getPath());
+        if (!directory.toFile().exists() || directory.toFile().listFiles().length == 0) {
           // If no snapshot files are created, snapshot is not valid
-          isValid = false;
+          abortInternal();
+          takenFuture.completeExceptionally(
+              new IllegalStateException(
```

Comment:
Does it make sense here to use also snapshot not found exception?

Reply:
I thought about it too. But didn't use it because it is still during "taking" a snapshot. So snapshot not found did not make sense to me.